pylint/checkers/spelling.py

```diff
@@ -310,17 +313,23 @@ def open(self) -> None:
         ]
 
         if self.linter.config.spelling_private_dict_file:
-            self.spelling_dict = enchant.DictWithPWL(
-                dict_name, self.linter.config.spelling_private_dict_file
-            )
+            self.spelling_dicts = [
+                enchant.DictWithPWL(
+                    dict_name, self.linter.config.spelling_private_dict_file
+                )
+                for dict_name in dict_names
+                if dict_name
+            ]
         else:
-            self.spelling_dict = enchant.Dict(dict_name)
+            self.spelling_dicts = [
+                enchant.Dict(dict_name) for dict_name in dict_names if dict_name
+            ]
 
         if self.linter.config.spelling_store_unknown_words:
             self.unknown_words: set[str] = set()
 
         self.tokenizer = get_tokenizer(
-            dict_name,
+            tokenization_language,
```

Comment:
Arbitrarily selecting the first configured locale to use as a tokenization locale is probably the part of this pull request that I'm most dubious about.

In defense of it, though: we only allow choice of _locales_, not choice of entirely separate _languages_ -- so `en_US, en_GB` is OK, but `en_GB, de_DE` is not valid here.

Do different locales within a language always share the same tokenization rules?  I'm not entirely qualified to say, but my guess from experience of these things is likely "no, there's some edge case locales" -- but hopefully it's relatively rare.

Reply:
See also [previous discussion](https://github.com/pylint-dev/pylint/issues/8083#issuecomment-1529446843) about this.